### PR TITLE
 Multiline var bracket next line support

### LINKED ISSUE #58: Validator: multiline `var` handling is broken
If `[` is placed at the next line, wrong alert is raised:

Selection_001

```elm
var bmcs =
  [
  "b_idx_above",
  "b_tail_below",
  "b_valmin",
  "b_win_greater"
  ]
endvar
```

### PR DESCRIPTION
**Pull request by PolinaShneider**
Closes #58 

image

### CODE REVIEW

File: src/test/var.test.ts [hunk 1]
```diff
@@ -1,78 +1,216 @@
...
+    test("Incorrect multiline var array", () => {
+        const config = baseConfig(`var v = [
+            [9,3], [9,4]
+        ]
+        edvar`);
+        const validator = new Validator(config);
+        const actualDiagnostic = validator.lineByLine();
+        const expectedDiagnostic = [
+            createDiagnostic(
+                Range.create(Position.create(7, 6), Position.create(7, 9)),
```
Comment (annstriganova) [inline src/test/var.test.ts]: `createRange` is less verbose.

File: src/test/var.test.ts [hunk 2]
```diff
@@ -1,78 +1,216 @@
...
+        const validator = new Validator(config);
+        const actualDiagnostic = validator.lineByLine();
+        const expectedDiagnostic = [];
+        deepStrictEqual(actualDiagnostic, expectedDiagnostic, `Config: \n${config}`);
+    });
+
+    test("Correct multiline var, bracket next line", () => {
+        const config = baseConfig(`var v =
+        [
+            "a"
```
Comment (annstriganova) [inline src/test/var.test.ts]: It seems that it's `Incorrect multiline var...`

File: src/test/var.test.ts [hunk 3]
```diff
@@ -1,78 +1,216 @@
...
+  ]
+endvar
+
+`;
+        const formatter = new Formatter(FORMATTING_OPTIONS);
+        const actual = formatter.format(text);
+        deepStrictEqual(actual, expected);
+    });
+
+    test("Incorrect multiline var, bracket next line", () => {
```
Comment (annstriganova) [inline src/test/var.test.ts]: `Correct multiline var, bracket next line`?

File: src/regExpressions.ts [hunk 4]
```diff
@@ -110,3 +110,15 @@ export const UNCLOSED_SECTION_DECLARATION: RegExp = /(^\s*\[)(\w+)\s*$/;
...
+export const VAR_OPEN_BRACKET_SAME_LINE: RegExp = /=\s*(\[|\{)(|.*,)\s*$/g;
+
+// var test =
+// [
+export const VAR_OPEN_BRACKET_NEXT_LINE: RegExp = /\s*(\[|\{)(|.*,)\s*$/g;
+
+// var test = [
+//     ...
+// ]  <- close bracket
+export const VAR_CLOSE_BRACKET: RegExp = /((\s*[\]\}\)]\s*)+)/g;
```
Comment (annstriganova) [inline src/regExpressions.ts]: `)` - I think it can not be closing bracket.

Comment (PolinaShneider) [reply src/regExpressions.ts]: According to validator, we support three kinds of brackets: `}`, `)`, `]`. However, formatter seems not to indent `)` correctly, even if it is located the same line as equal sign.

Do we need `)` parentheses support in multiline var?

Comment (annstriganova) [reply src/regExpressions.ts]: I was wrong. [`var`](https://axibase.com/docs/charts/syntax/control-structures.html#var) in Charts is equal to `var` in JS. It seems that the following construction is validated based on `()`:

```js
var hosts = getEntities(
  'svl-hosts')

var hosts = getEntities('svl-hosts')
```
Don't forget `(` in `VAR_OPEN_BRACKET...`.

**Comment by PolinaShneider** [comment]
Couldn't save history, had to [reopen](https://github.com/axibase/charts-language-service/pull/60) PR and perform cherry-pick. Will correct code according to comments now

File: src/test/var.test.ts (hunk 2 above)
Comment (PolinaShneider) [reply src/test/var.test.ts]: No. First of all, indent of `endvar` should be smaller of `]`. Secondly, in incorrect test original config cannot be **identical** to expected.

Comment (annstriganova) [reply src/test/var.test.ts]: Ok, there is no comma after `"a"`, it baffled me. Could you rename tests, please, use more specific description, for example:
* `Correct multiline var: var is declared, bracket next line`
* `Incorrect multiline var: no var is declared, bracket next line`
* ...
I mean, make emphasis, why `var` is correct. Also it's convenient to create `suite("Bracket next line")`.

**Pull request closed without merging**